x-pack/plugin/ilm/src/test/java/org/elasticsearch/xpack/slm/SLMSnapshotBlockingIntegTests.java

```diff
@@ -257,26 +257,17 @@ public void testRetentionWhileSnapshotInProgress() throws Exception {
         }
     }
 
-    public void testBasicFailureRetention() throws Exception {
```

Comment:
should we still test this scenario? Can folks disable partial snapshots on SLM?

Reply:
I don't think we have to, there's no more `FAILED` state snapshots being created in the repo with this change (not you oculd only ever get a `FAILED` snapshot if partial was turned off in the first place). We could force the creation of a `FAILED` snapshot as a BWC test maybe by manually messing with the `RepositoryData` (come to think of it ... I'll do that, otherwise we lose coverage for a BwC scenario).

Reply:
This needs some trickier tests it turns out but is well worth it given how SLM low-level manages these snapshot states. I opened https://github.com/elastic/elasticsearch/pull/59082 to enable the necessary test infrastructure.